### DISCUSSION TITLE
Copilot 'Tab to accept' not working in Visual Studio C++

### DISCUSSION THREAD

**Discussion by rtischer8277** [community/community#44589] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Brand new Copilot user. My editor is Visual Studio C++. My keyboard mapping scheme is **Visual C++ 6**. The code suggestions look interesting, but accepting is not working for my setup. 

This online help called [Trigger inline suggestion](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-github-copilot-in-visual-studio#rebinding-keyboard-shortcuts) says to use a different key combo. That didn't work either.

Tools->Options->Environment->Keyboard confirms, that for **Trigger inline suggestion** the key combo should be **Ctrl+Alt+\\** for command name **Edit.Copilot.TriggerInlineSuggestion**. But that doesn't work either.

Does anyone have Copilot experience with a setup like mine?

**Comment by rtischer8277** [comment]
**Tab to accept** comments now work with Version 17.5.0 Preview 3.0. It still does not work for accepting code suggestions though.

**Reply by GGabc** [reply]
Isn't the point of Copilot to provide code, not comments?  Sounds like Copilot is pretty %/&* useless.

**Reply by rtischer8277** [reply]
I am thinking that the 'Tab to accept' doesn't work because I have some other accelerator key set interfering with it like Google. I'm just not willing to put time on figuring that out. That's MS's job. I do like watching the code suggestions though. They will get better over time I'm sure once the dataset gets better.

**Comment by rtischer8277** [comment]
BTW, this latest edition of VS2022 Preview has gotten rid of the 'Tab to accept'. Now it offers up a line of code, I start typing in the code if I think it will work, when I come to a variable, there a dropdown list which I can choose from, all the while keeping on typing, and it worked. Here is the what was suggested and what worked the first time:
			// Load the resource from its path name.
			HBITMAP hBitmap = ( HBITMAP )LoadImage( NULL, retrievedThisComputerImagePathName, IMAGE_BITMAP, 0, 0, LR_LOADFROMFILE );
The variable name was taken from the line above where I had retrieved the file path of the resource. 
Very very slick.
